src/stcal/alignment/util.py

```diff
@@ -733,17 +721,17 @@ def update_s_region_keyword(model, footprint):
         String containing the S_REGION object.
     """
     s_region = "POLYGON ICRS  {:.9f} {:.9f} {:.9f} {:.9f} {:.9f} {:.9f} {:.9f} {:.9f}".format(
-        *footprint.flatten()
+       *footprint.flatten()
     )
     if "nan" in s_region:
         # do not update s_region if there are NaNs.
         log.info("There are NaNs in s_region, S_REGION not updated.")
-    else:
-        model.meta.wcsinfo.s_region = s_region
-        log.info("Update S_REGION to %s", model.meta.wcsinfo.s_region)
+        return None
+    log.info("Update S_REGION to %s", s_region)
+    return s_region
 
 
-def reproject(wcs1, wcs2):
+def reproject(wcs1: gwcs.wcs.WCS, wcs2: gwcs.wcs.WCS) -> Callable:
```

Comment:
I think this function needs to replicate what the[ jwst version](https://jira.stsci.edu/browse/RSOCDR-145) does, which is the most general way. The use case that this is missing is passing `sliced_wcs.SlicedLowLevelWCS` object. It's a JWST use case, that was requested by external users. There are tests for this in jwst:
https://github.com/spacetelescope/jwst/blob/master/jwst/resample/tests/test_utils.py#L191
Were they not moved to stcal or am I not seeing them?

Reply:
@nden Thanks for the comment.  I think the first link goes to something unrelated, unless I'm missing something.  There are two `reproject` functions in `jwst`, one in `assign_wcs.util` and one in `resample.resample_utils`.  The one in `resample` is not affected by this PR, and is the one that those tests cover.

In my opinion, consolidating these functions falls under the scope of [this issue](https://github.com/spacetelescope/stcal/issues/271) and does not need to be done here.  But I can work on changing it if you like